ml_metadata/tools/mlmd_bench/fill_types_workload.cc

```diff
@@ -84,42 +206,69 @@ tensorflow::Status FillTypes::SetUpImpl(MetadataStore* store) {
   // created.
   std::minstd_rand0 gen(absl::ToUnixMillis(absl::Now()));
 
-  // TODO(briansong): Add update support.
-  for (int64 i = 0; i < num_operations_; i++) {
+  // Gets the number of current types(`num_curr_type`) and total
+  // types(`num_total_type`) for later insert or update.
+  int64 num_curr_type = 0, num_total_type = 0;
+  GetTypesResponseType get_response;
+  TF_RETURN_IF_ERROR(GetNumberOfTypes(fill_types_config_, store, num_curr_type,
```

Comment:
A better way to support the update probably is reusing the previous code, and simply add the following lines:

1. get existing types  `existing_types`
ReadExistingTypes ..

2. keep the original for loop (split MakeUpTypesForUpdate part to each item)

```
+ absl::optional<T> maybe_existing_type = PrepareType<T>(config_, i, existing_types)
   GenerateType<ArtifactType>(
      type_name, num_properties, 
+    maybe_existing_type, 
      *absl::get<PutArtifactTypeRequest>(put_request).mutable_artifact_type(), 
      curr_bytes);
```

for PrepareType
```
// returns a absl::nullopt if config_ is not update. 
// otherwise, it either returns an existing type, or inserts a new one and returns it. 
absl::optional<T> PrepareType<T>(config_, i, existing_types) {
    if (!config_.update()) {
        return absl::nullopt; 
    }
    if (i < existing_types.size()) {
        return existing_types[i]
   }
   // creates a new type and return 
   existing_types->emplace_back() ..
   store->PutXyzType(existing_types.back())
   return type ..
}

```

Reply:
// unit-level util functions
void SetValue<T>(const T& type, variant_var);
void SetValue<ArtifactType>(const ArtifactType& type, variant_var) {
}
void SetValue<ExecutionType>(const ExecutionType& type, variant_var) {
}


T1 := union {ArtifactType, ExecutionType, ContextType}

vector<T1> existing_types; 
store->PutArtifactTypes()
store->PutTypes()

Reply:
Already refactored the code. Please look at the newest PR. Thanks.